mujoco_warp/_src/util_misc.py

```diff
@@ -0,0 +1,386 @@
+# Copyright 2025 The Newton Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+"""Miscellaneous utilities."""
+
+from typing import Tuple
+
+import warp as wp
+
+from . import math
+from .types import MJ_MINVAL
+from .types import WrapType
+
+
+@wp.func
+def is_intersect(p1: wp.vec2, p2: wp.vec2, p3: wp.vec2, p4: wp.vec2) -> bool:
```

Comment:
We haven't been great about this but let's add some docstrings.  It's on my list to add more as well.  So for example, this could read:

```py
def is_intersect(p1: wp.vec2, p2: wp.vec2, p3: wp.vec2, p4: wp.vec2) -> bool:
  """Check for intersection of two 2D line segments."""
```

Don't need to be super long, and feel free to crib the comment strings from MJC `engine_util_misc.c`

Reply:
done